ChopChop/Models/Ingredients/Ingredient.swift

```diff
@@ -0,0 +1,271 @@
+import Foundation
+import GRDB
+
+/**
+ Represents an ingredient, consisting of batches grouped by expiry date.
+ 
+ Invariants:
+ - All quantities are of the same type.
+ - Each batch has a unique expiry date.
+ */
+class Ingredient: FetchableRecord {
+    var id: Int64?
```

Comment:
Making the `id`s `private(set) var` would probably be better.
@seowalex Can the `id` be non-optional?

Reply:
It's optional because when you first create an Ingredient, it doesn't have an `id`. It'll only be populated by the save methods after saving, which it's why it's variable.